tests/integration/mock/MonologueAgent/test_write_simple_script/prompt_001.log

```diff
@@ -318,6 +318,18 @@ This is your internal monologue, in JSON format:
       "content": "It seems like there might be an existing project here. I should probably start by running `pwd` and `ls` to orient myself.",
       "wait_for_response": false
     }
+  },
```

Comment:
Does the test fail without modifying these test artifacts?

Reply:
I believe the latest `regenerate.sh` file on main doesn't regenerate test artifacts if the tests are passing.

Reply:
Re: inheritance, I added some more notes to the docstring on Plan.

Ultimately, we have a tree structure representing the agent's plan, e.g.

1. Do x
  * Do x'
  * Do x''
    * Do x''1
    * Do x''2
3. Do y
  * Do y'
  * Do y''
4. Do z 

What's odd about this data structure is that the top-level is a _list_, not a single node. So to hold onto it, we need a sort of invisible root node. That's where Plan comes in.

Plan needs to be the same type as every other node, so that `parent` references work nicely. But it's also a special node that doesn't contain any information itself, hence the subclassing.